packages/x-data-grid/src/models/gridBaseSlots.ts

```diff
@@ -1,10 +1,35 @@
+type Ref = React.RefCallback<HTMLElement> | React.RefObject<HTMLElement> | null;
+
 export type BadgeProps = {
   badgeContent?: React.ReactNode;
   children: React.ReactNode;
   color?: 'primary' | 'default' | 'error';
+  invisible?: boolean;
   overlap?: 'circular';
   variant?: 'dot';
-  invisible?: boolean;
+  style?: React.CSSProperties;
+};
+
+export type ButtonProps = {
+  ref?: Ref;
+  children?: React.ReactNode;
+  className?: string;
+  disabled?: boolean;
+  id?: string;
+  onClick?: React.MouseEventHandler<HTMLElement>;
+  onKeyDown?: React.KeyboardEventHandler<HTMLElement>;
+  role?: string;
+  size?: 'small' | 'medium' | 'large';
+  startIcon?: React.ReactNode;
+  style?: React.CSSProperties;
+  tabIndex?: number;
+  title?: string;
+  touchRippleRef?: any; // FIXME(v8:romgrk): find a way to remove
+};
+
+export type IconButtonProps = Omit<ButtonProps, 'startIcon'> & {
+  color?: 'default' | 'primary';
+  edge?: 'start' | 'end' | false;
```

Comment:
There are quite a few props for buttons that don't make sense in a non-Material UI world, including this, but I think we can fine tune these later.

Reply:
You mean `edge`? Yeah I guess so, though if we remove it I'm sure it's going to cause a visual regression somewhere. I'm still wondering how we're going to add those little tunings while being design-system agnostic.